TrafficCapture/dockerSolution/src/main/docker/migrationConsole/workflows/captureReplay.yaml

```diff
@@ -0,0 +1,196 @@
+apiVersion: argoproj.io/v1alpha1
+kind: WorkflowTemplate
+metadata:
+  name: capture-replay
+spec:
+  entrypoint: run-all
+  serviceAccountName: argo-workflow-executor
+  templates:
+    - name: run-all
+      failFast: true
+      inputs:
+        parameters:
+          - name: session-name
+            # temporary - just to launch a migration console
+          - name: source-config
+            value: ""
+            # Kafka setup
+          - name: provided-kafka-bootstrap-servers
+            value: ""
+          - name: provided-kafka-k8s-name
+            value: ""
+          - name: kafka-prefix
+            value: "ct"
+          - name: topic-name
+            value: ""
+          - name: topic-partitions
+            value: ""
+            # Proxy setup
+          - name: proxy-destination
+          - name: proxy-listen-port
+            # Replayer setup
+          - name: replayer-target-config
+      dag:
+        tasks:
+          - name: id-generator
+            template: id-generator
+            arguments:
+              parameters:
+                - name: service-name
+                  value: "{{inputs.parameters.session-name}}"
+                - name: proxy-endpoint
+                  value: "http://{{inputs.parameters.session-name}}:{{inputs.parameters.proxy-listen-port}}"
+
+          - name: kafka-cluster-setup
+            templateRef:
+              name: kafka-setup
+              template: cluster-deploy
+            when: "'{{inputs.parameters.provided-kafka-bootstrap-servers}}' == ''"
+            dependencies: [ id-generator ]
+            arguments:
+              parameters:
+                - name: kafka-name
+                  value: "{{inputs.parameters.kafka-prefix}}-{{=last(split(tasks['id-generator'].id, '-'))}}"
```

Comment:
How can we validate this value is constructed correctly?  What happens if we get the task reference wrong?